### PR TITLE
Add step components

### PR DESCRIPTION
**Pull request by LizBaker**
you can check them out in an mdx page in the demo at `/steps`

Screen Shot 2023-01-18 at 4 15 05 PM

### CODE REVIEW

**Review by tabathadelane — COMMENTED** [review]
:danceydoge::

**Review by tabathadelane — DISMISSED** [review]

**Comment by roadlittledawn** [comment]
should we add docs to [README](https://github.com/newrelic/gatsby-theme-newrelic/blob/liz/add-step-components/packages/gatsby-theme-newrelic/README.md) for `Steps` and `Step`?

**Comment by roadlittledawn** [comment]
when clicking `Copy` for a code block, it seems to temporarily expand the main column? not sure if this is a layout issue with demo or new components ¯\_(ツ)_/¯

https://user-images.githubusercontent.com/2952843/213543690-9a034198-507d-4df4-b292-4d8c84f11c61.mp4

File: packages/gatsby-theme-newrelic/src/components/TutorialSteps/TutorialStep.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import { isMdxType } from '../../utils/mdx';
+
+const ALLOWED_HEADINGS = ['h2', 'h3', 'h4'];
+
+const TutorialStep = ({ children, stepNumber, totalSteps }) => {
+  children = Children.toArray(children);
+
+  const title = ALLOWED_HEADINGS.some((level) => isMdxType(children[0], level))
+    ? children[0]
+    : null;
```
Comment (roadlittledawn) [inline packages/gatsby-theme-newrelic/src/components/TutorialSteps/TutorialStep.js:13]: for some reason i'm not super clear on what we're doing here or if it's necessary? if anything, might be worth documenting for writers if they want to set a section title.

if the immediate child of the `Step` component is a `<h2>`, `<h3>`, `<h4>` set it as the title of the section. if not, do nothing and move on.

weirdly, if you use markdown syntax for immediate child like so:

```
<Step>
## Heading 2

More content
```

results in this:



but doing this:

```
<Step>
<h2>Heading 2</h2>

More content
```

results in this, which is more like what's expected?

**Comment by clarkmcadoo** [comment]
> when clicking `Copy` for a code block, it seems to temporarily expand the main column? not sure if this is a layout issue with demo or new components ¯_(ツ)_/¯
> 
>  2023-01-19_11-40-48.mp4

I _think_ this is just how that button and spacing is being handled, it doesn't occur on the lower code block

**Review by clarkmcadoo — DISMISSED** [review]

**Review by roadlittledawn — DISMISSED** [review]
Minor comments. We probably want to at least add doc to README

**Comment by LizBaker** [comment]
for the copy button situation i did find a little bug in SideBySide which i can fix but otherwise unsure if the steps have anything to do with that

as far as refactoring the existing code goes we can do that now but it will slow us down and it means transferring the docs might not be 1:1 copy/paste for the writers 🤷‍♀️ not sure what the best path is

**Review by clarkmcadoo — APPROVED** [review]

**Pull request merged**